src/function/list/list_reduce.cpp

```diff
@@ -22,12 +22,28 @@ static std::unique_ptr<FunctionBindData> bindFunc(ScalarBindFuncInput input) {
         ListType::getChildType(input.arguments[0]->getDataType()).copy());
 }
 
+static uint32_t findIndex(const std::vector<std::string>& strings, const std::string& str) {
+    auto it = std::find(strings.begin(), strings.end(), str);
+    if (it != strings.end()) {
+        return it - strings.begin();
+    } else {
+        throw common::RuntimeException{"list_reduce lambda var name cannot found."};
+    }
+}
+
 static void reduceList(const list_entry_t& listEntry, uint64_t pos, common::ValueVector& result,
     common::ValueVector& inputDataVector, const common::ValueVector& tmpResultVector,
     evaluator::ListLambdaBindData& bindData) {
-    auto xParam = bindData.lambdaParamEvaluators[0]->resultVector.get();
-    auto yParam = bindData.lambdaParamEvaluators[1]->resultVector.get();
-    auto paramPos = xParam->state->getSelVector()[0];
+    std::vector<ValueVector*> params(bindData.lambdaParamEvaluators.size());
+    std::vector<uint32_t> indexInVars(bindData.lambdaParamEvaluators.size());
+    for (uint32_t i = 0; i < bindData.lambdaParamEvaluators.size(); i++) {
```

Comment:
The idx of parameter should be resolved at compile time because it will never change.

I would either move this logic to `bindFunc` and add `indexInVars` in `ListLambdaBindData`. 

Alternatively you can also consider making this a general logic for ListLambdaEvaluator, meaning not only `list_reduce`, `list_transform` and `list_filter` should also try to resolve a idx based on parameter name. Either way sounds reasonable to me.

Reply:
Thanks for your reply.  I made the modifications and submitted a new commit for your suggestion.